hudi-test-suite/src/main/java/org/apache/hudi/testsuite/generator/DeltaGenerator.java

```diff
@@ -108,6 +114,17 @@ public DeltaGenerator(DeltaConfig deltaOutputConfig, JavaSparkContext jsc, Spark
     return inputBatch;
   }
 
+  public JavaRDD<GenericRecord> generateUpsertsWithDistributedSource(Config operation) {
+    TypedProperties props = new TypedProperties();
+    props.setProperty(TestSourceConfig.MAX_UNIQUE_RECORDS_PROP, String.valueOf(operation.getNumRecordsInsert()));
+    props.setProperty(TestSourceConfig.NUM_SOURCE_PARTITIONS_PROP, String.valueOf(operation.getNumInsertPartitions()));
+    props.setProperty(TestSourceConfig.USE_ROCKSDB_FOR_TEST_DATAGEN_KEYS, "true");
+    DistributedTestDataSource distributedTestDataSource = new DistributedTestDataSource(
```

Comment:
Still see "test" names in the core logic - either rename it and add it to a utils folder so can be used in the src code.